File: GiftIdeaTracker/src/GiftIdeaTracker.Core/Model/RecipientAggregate/Recipient.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace GiftIdeaTracker.Core;

public class Recipient
{
    public Guid RecipientId { get; set; }
    public Guid UserId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Relationship { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<GiftIdea> GiftIdeas { get; set; } = new List<GiftIdea>();
}

File: GiftIdeaTracker/src/GiftIdeaTracker.WebApp/projects/gift-idea-tracker-shared/src/lib/models/recipient.ts
import { GiftIdea } from './gift-idea';

export interface Recipient {
  recipientId: string;
  userId: string;
  name: string;
  relationship: string | null;
  createdAt: string;
  giftIdeas?: GiftIdea[];
}

export interface CreateRecipientRequest {
  name: string;
  relationship?: string;
}

export interface UpdateRecipientRequest {
  recipientId: string;
  name: string;
  relationship?: string;
}
